comp/process/processcollector/processcollectorimpl/process_collector.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package processcollectorimpl implements the local process collector for Workloadmeta.
+package processcollectorimpl
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/benbjohnson/clock"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	proccontainers "github.com/DataDog/datadog-agent/pkg/process/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "local-process-collector"
+	componentName     = "workloadmeta-process"
+	cacheValidityNoRT = 2 * time.Second
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	wlmExtractor  *processwlm.WorkloadMetaExtractor
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+
+	// only used when process checks are disabled
+	processData       *Data
+	pidToCid          map[int]string
+	collectionClock   clock.Clock
+	containerProvider proccontainers.ContainerProvider
+}
+
+// NewCollector returns a new local process collector provider and an error.
+// Currently, this is only used on Linux when language detection and run in core agent are enabled.
+func NewCollector() (workloadmeta.CollectorProvider, error) {
```

Comment:
Please use the convention described in the doc https://datadoghq.dev/datadog-agent/components/creating-components (`Requires` and `Provides` struct, `NewComponent` function name ...).